internal/util/password/scramsha256.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package password
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+
+	"github.com/xdg-go/scram"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// scramSHA256Params represent password parameters for SCRAM-SHA-256 authentication.
+type scramSHA256Params struct {
+	iterationCount int
+	saltLen        int
+}
+
+// fixedScramSHA256Params represent fixed password parameters for SCRAM-SHA-256 authentication.
+var fixedScramSHA256Params = &scramSHA256Params{
+	iterationCount: 15_000,
+	saltLen:        45,
+}
+
+// scramSHA256HashParams hashes the password with the given salt and parameters,
+// and returns the document that should be stored.
+func scramSHA256HashParams(username, password string, salt []byte, params *scramSHA256Params) (*types.Document, error) {
+	if len(salt) != int(params.saltLen) {
+		return nil, lazyerrors.Errorf("unexpected salt length: %d", len(salt))
```

Comment:
Salt is created from `scramSHA256Params.saltLen`, so how could this happen?

Reply:
@AlekSi , what do you think about removing this assertion and the one for PLAIN as well? Or should we keep it as a safeguard as it's something critical?

Reply:
Just saying for me either is fine. Let the stronger voice speak :hugs: